drivers/eaton-pdu-marlin-mib.c

```diff
@@ -801,10 +810,14 @@ static snmp_info_t eaton_marlin_mib[] = {
 	{ "outlet.desc", ST_FLAG_RW | ST_FLAG_STRING, 20, NULL, "All outlets",
 		SU_FLAG_STATIC | SU_FLAG_ABSENT | SU_FLAG_OK, NULL },
 	/* UnitType
-	 * used to depict the overall outlets switchability of the unit */
+	 * used to depict the overall outlets switchability of the unit on G3 and newer ePDU*/
 	{ "outlet.switchable", ST_FLAG_STRING, SU_INFOSIZE,
 		".1.3.6.1.4.1.534.6.6.7.1.2.1.10.%i",
-		"no", 0, &marlin_unit_switchability_info[0] },
+		"no", SU_FLAG_STATIC | SU_FLAG_UNIQUE, &marlin_unit_switchability_info[0] },
+	/* Ugly hack for older G2 ePDU: check the first outlet to determine unit switchability */
+	{ "outlet.switchable", ST_FLAG_STRING, SU_INFOSIZE,
+		".1.3.6.1.4.1.534.6.6.7.6.6.1.3.%i.1",
+		"no", SU_FLAG_STATIC | SU_FLAG_UNIQUE | SU_OUTLET | SU_FLAG_OK | SU_TYPE_DAISY_1, &g2_unit_outlet_switchability_info[0] },
```

Comment:
Do I get this right that as the driver walks a device, it first hits the OID several lines above and sets the dstate value by marlin_unit_switchability_info mapping, then hits this OID too and re-sets the dstate by g2_unit_outlet_switchability_info?

Does not sound *too* ugly ;) Might even be upstreamable if that works too :)

Reply:
not exactly for the behavior:
* on G2, the first outlet.switchable doesn't exist, so try the 2nd, which is the ugly hack: try on the first outlet to see if switchable
* on G3, the first is good, so flag unique does the rest

can be upstreamed indeed, though not enough time on my side as usual